play-json/shared/src/main/scala/Json.scala

```diff
@@ -414,6 +419,29 @@ object Json extends JsonFacade {
   sealed trait MacroOptions
 
   /**
+   * Defines the default macro options.
+   *
+   * Since you can't have defaults for a type parameter (unless it's a contravariant type, it will default to
+   * Nothing), we supply the default via an implicit parameter.
+   */
+  trait DefaultMacroOptions[O <: Json.MacroOptions]
+
+  /**
+   * Low priority implicit used when Json.MacroOptions is not passed.
```

Comment:
`JsonConfiguration.apply` now takes a `DefaultMacroOptions` type class, so an implicit for every type that you want to pass to it must be defined. The highest priority implicit is `DefaultMacroOptions[MacroOptions]`, this causes `MacroOptions` to be the default if nothing is explicitly specified. But if something that is not the default gets passed, then we also need to provide a `DefaultMacroOptions` implicit for it, that is defined on this trait, and it has to be in a trait to make it lower priority than the default.